### PR TITLE
support kafka -> structured streaming data process

### PR DESCRIPTION
**Pull request by riverzzz**
support auto transfor kafka json data to df by flields,and keep field data type unchanged
support set kafka startingOffsets and maxOffsetsPerTrigger
split data check and transfer process logic,split it to filter and map;change error streaming data process logic,will only print log 
pom add streaming process dependencies,exclusion unnecessary dependencies

### CODE REVIEW

**Comment by Nicole00** [comment]
Thanks for your contribution~ That's exactly what I'm going to do.

Besides to skip the invalid data, can we extract the data pre-process to a function?

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala [hunk 1]
```diff
@@ -36,18 +35,24 @@ class KafkaReader(override val session: SparkSession,
...
+        .option("subscribe", kafkaConfig.topic)
+        .option("startingOffsets", kafkaConfig.startingOffsets)
+
+    val maxOffsetsPerTrigger = kafkaConfig.maxOffsetsPerTrigger
+    if(maxOffsetsPerTrigger.isDefined) reader.option("maxOffsetsPerTrigger", maxOffsetsPerTrigger.get)
+
+    reader.load()
+      .select($"value".cast(StringType))
+      .select($"value", json_tuple($"value", fields: _*))
+      .toDF("value" :: fields: _*)
```
Comment (Nicole00) [inline nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala:54]: There's no need to keep the `value` field in the dataframe.

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/config/Configs.scala [hunk 2]
```diff
@@ -249,6 +249,7 @@ object Configs {
   private[this] val DEFAULT_LOCAL_PATH           = None
   private[this] val DEFAULT_REMOTE_PATH          = None
   private[this] val DEFAULT_STREAM_INTERVAL      = 30
+  private[this] val DEFAULT_KAFKA_STARTINGOFFSETS      = "latest"
```
Comment (Nicole00) [inline nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/config/Configs.scala:252]: format the code style please

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/config/SourceConfigs.scala [hunk 3]
```diff
@@ -162,16 +162,20 @@ case class MySQLSourceConfigEntry(override val category: SourceCategory.Value,
   *
   * @param server
   * @param topic
+ * @param startingOffsets
```
Comment (Nicole00) [inline nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/config/SourceConfigs.scala:165]: format the code style please

File: nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala (hunk 1 above)
Comment (riverzzz) [reply nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala:54]: I suggest keep it,if zhe data is invalid data,i will know source data to help fix bug

Comment (Nicole00) [reply nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala:54]: there's no print operation for df, how can we know the source data's format? 
the way is to debug the data, then can we use `watch` to get the value data ?

Comment (riverzzz) [reply nebula-exchange/src/main/scala/com/vesoft/nebula/exchange/reader/StreamingBaseReader.scala:54]: in Processor.scala filter will print it
i don't know watch means

**Comment by Nicole00** [comment]
please rebase the master code and rerun the ci please.

**Review by Nicole00 — APPROVED** [review]
LGTM! Great work, thanks again

**Pull request merged**